Firestore/core/src/firebase/firestore/api/query_core.h

```diff
@@ -167,6 +167,18 @@ class Query {
   void ValidateNewOrderByPath(const model::FieldPath& fieldPath) const;
   void ValidateOrderByField(const model::FieldPath& orderByField,
                             const model::FieldPath& inequalityField) const;
+  /**
+   * Validates that the value passed into a disjunctive filter satisfies all
+   * array requirements.
+   */
+  void ValidateDisjunctiveFilterElements(const model::FieldValue& field_value,
+                                         const core::Filter::Operator op) const;
+
+  model::FieldValue ParseDocumentIdValue(
```

Comment:
Please add a comment describing what this does. From its name I'm not really sure what it does at all.

Reply:
Given this comment and the way you're using it. I'm not entirely sure I understand why this name is associated with "DocumentId". Your description doesn't mention it. Fundamentally this name is problematic because there's never really such a thing as a "document ID value". Document IDs are associated with the field of a filter, not the value.

Some ideas for how to resolve this (assuming I'm even understanding correctly):

1) Since your description is "Parses the given FieldValue into a Reference", shouldn't this name be something like ParseReferenceValue or maybe ParseExpectedReferenceValue?

2) Or maybe the issue is that we're parsing a reference value because we're expecting that whenever you supply a value for a document ID field? In that case you might consider naming this ParseValueForDocumentIdFilter or something like that.

Reply:
Thanks for breaking it down. The intention is that this method is called to validate that a provided FieldValue is indeed a valid Reference value, since References can also be Strings. 

In that case, it makes sense to call it `ParseExpectedReferenceValue`, since we are parsing that an "expected" Reference value is actually a valid Reference value.